src/hotspot/share/utilities/count_trailing_zeros.hpp

```diff
@@ -59,13 +65,23 @@ inline unsigned count_trailing_zeros(uintx x) {
 #pragma intrinsic(_BitScanForward)
 #endif
 
-inline unsigned count_trailing_zeros(uintx x) {
-  assert(x != 0, "precondition");
+inline unsigned count_trailing_zeros_32(uint32_t x) {
+  unsigned long index;
+  _BitScanForward(&index, x);
+  return index;
+}
+
+inline unsigned count_trailing_zeros_64(uint64_t x) {
   unsigned long index;
 #ifdef _LP64
   _BitScanForward64(&index, x);
 #else
-  _BitScanForward(&index, x);
+  if (_BitScanForward(&index, (uint32_t)x) == 0) {
```

Comment:
Is the cast needed?  That is, does the implicit conversion from uint64_t to (32bit) unsigned long not work?  If it's needed, please use static_cast.